app/assets/stylesheets/govuk_publishing_components/components/_action-link.scss

```
@@ -1,15 +1,13 @@
 .gem-c-action-link {
   display: table;
 
-  &:before {
+  &::before {
```

The double colon pseudo-element isn't supported in IE 8 - whilst we don't support IE8, I think it'd be worthwhile keeping it as a single colon to give us time to check what effect that this could have so that we can ensure that it doesn't break anything.

```suggestion
  &:before {
```

---

Thanks @injms for taking a look and the feedback. Do you think this is a change all of GDS should have or just this repo?

This rule is an area of discussion to have with the design-system too as they have a mixture of single and double colons usage in their codebase. They, however, use autoprefixer which replaces the double ones with single ones anyway so their IE8 stylesheets are generated with single colons.

If we don't support IE8 do we need to worry about maintaining support for one of its features? I had a quick look in IE8 on gov.uk homepage and can see a few things broken so I'd wonder if this would be an overly-cautious move that blurs the lines of support and non-support.

---

We don't support IE8, but for the sake of an extra colon it doesn't seem worth potentially making IE8 users experience of GOV.UK worse. There's still something like 40,000 IE8 users (at last count, admittedly from nearly a year ago) but that's a lot of people. GOV.UK looks quite broken in IE8, but it should still be usable.

---

I think that this is a change that all of GDS should have - so can put that into a comment on the linter itself if that helps? 

Changing from single to double is a change that has no effect _apart_ from breaking more things in IE 8. The support for both single and double is the same - and will continue to be so as browsers have to keep supporting both. 

In the spirit of users > devs > technical purity, because this is a change that only breaks things for some of the users stuck on IE8,  and because this doesn't fix or improve anything apart from the code - I think single colon pseudo elements should be the default.

---

Sure thanks for explaining, I'll flip this round in the rules to be for single colon. If that then proves divisive we can discuss further.

---

@injms or @andysellick, I've just opened https://github.com/alphagov/stylelint-config-gds/pull/1 would one of you mind reviewing?

---

I've switched this around now @injms

---

Super - thanks for making the changes.